lib/ansible/modules/cloud/amazon/ec2_vpc_vgw_facts.py

```diff
@@ -0,0 +1,168 @@
+    required: false
+    default: None
+  vpn_gateway_ids:
+    description:
+      - Get details of a specific Virtual Gateway ID. This value should be provided as a list. 
+    required: false
+    default: None
+author: "Nick Aslanidis (@naslanidis)"
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# # Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Gather facts about all virtual gateways for an account or profile
+  ec2_vpc_vgw_facts:
+    region: ap-southeast-2
+    profile: production
+  register: vgw_facts
+
+- name: Gather facts about a filtered list of Virtual Gateways
+  ec2_vpc_vgw_facts:
+    region: ap-southeast-2
+    profile: production
+    filters:
+        "tag:Name": "vgw-123"
+  register: vgw_facts
+
+- name: Gather facts about a specific virtual gateway by VpnGatewayIds
+  ec2_vpc_vgw_facts:
+    region: ap-southeast-2
+    profile: production
+    vpn_gateway_ids: vgw-c432f6a7 
+  register: vgw_facts
+'''
+
+RETURN = '''
+virtual_gateways:
+    description: The virtual gateways for the account.
+    returned: always
+    type: list
+    sample: [
+        {
+            "state": "available",
+            "tags": [
+                {
+                    "key": "Name",
+                    "value": "TEST-VGW"
+                }
+            ], 
+            "type": "ipsec.1",
+            "vpc_attachments": [
+                {
+                    "state": "attached",
+                    "vpc_id": "vpc-22a93c74"
+                }
+            ],
+            "vpn_gateway_id": "vgw-23e3d64e"
+        }
+    ]
+
+changed:
+    description: True if listing the virtual gateways succeeds.
+    returned: always
+    type: bool
+    sample: "false"
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be captured by imported HAS_BOTO3
+
+
+def get_virtual_gateway_info(virtual_gateway):
+    virtual_gateway_info = {'VpnGatewayId': virtual_gateway['VpnGatewayId'],
+                             'State': virtual_gateway['State'],
+                             'Type': virtual_gateway['Type'],
+                             'VpcAttachments': virtual_gateway['VpcAttachments'],
+                             'Tags': virtual_gateway['Tags']}
+    return virtual_gateway_info
+
+
+def list_virtual_gateways(client, module):
+    params = dict()
+
+    params['Filters'] = ansible_dict_to_boto3_filter_list(module.params.get('filters'))
+    params['DryRun'] = module.check_mode
+
+    if module.params.get("vpn_gateway_ids"):
+        params['VpnGatewayIds'] = module.params.get("vpn_gateway_ids")
+
+    try:
+        all_virtual_gateways = client.describe_vpn_gateways(**params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
```

Comment:
Can you add the exception path here, with `fail_json(msg=...., exception=traceback.format_ext())`